README.md

```diff
@@ -1,33 +1,107 @@
-* To have test cases for end-user backend tools such as dfx, sns-quill, etc.
-* To test upcoming UX improvements to the [NNS frontend dapp](https://nns.ic0.app/) before releasing them.
+## Purpose
 
-Note. This repository has not been tested with Apple M1 devices!
+The main purpose of `sns-testing` is to enable developers of Internet Computer (IC) dapps to test Service Nervous System (SNS) decentralization. However, this solution may also be applicable in other scenarios, e.g.:
 
-## Bootstrapping a testing environment
+* Testing tools such as `dfx`, `sns-quill`.
+* Testing UX-related aspects before releasing the [NNS frontend dapp](https://nns.ic0.app/).
 
-This section explains the simplest way to set up a local testing environment and run a basic scenario deploying a test canister to a local replica and handing over control of this canister to an SNS. This basic scenario will help you better understand all the steps involved in handing over control of your own canister to an SNS; please refer to [SNS lifecycle](https://github.com/dfinity/sns-testing#sns-lifecycle) for more details.
+## How to use these instructions
+
+Assuming you are a developer who wants to test SNS-decentralization of their dapp, you need to establish your own dapp's deployment process before using this solution. Usually, a testing deployment is done in a shell script that interacts with a local replica via [DFX](https://internetcomputer.org/docs/current/references/cli-reference/dfx-parent).
+
+You might need to slightly adjust your deployment script to work with sns-testing. In particular, please avoid running `dfx start` or `dfx replica` inside your deployment script (`sns-testing` will take care of starting a replica instance for you).
+
+If you do not yet have a dapp that is ready for decentralization, you may still run `sns-testing` with the built-in example dapp.
+
+## Special instructions for Apple silicon users
+
+<a name="apple-silicon"></a>
+
+_[Skip to the next section](#docker) if you are using an x86-compatible system, e.g., Linux, Windows, or Intel-based Mac._
+
+The `sns-testing` solution is based on Docker; however, there are subtle issues while running Docker on new [Apple silicon](https://support.apple.com/en-us/HT211814) systems (e.g., Apple M1, Apple M2). Therefore, Apple silicon users are advised to run the commands provided by this repository _directly_. This requires additional preparation:
+
+1. Make sure you have Homebrew installed.
+   * Instructions: https://brew.sh/
+2. Install `coreutils` (needed for tools e.g., `sha256sum`) and `jq`:
+   ```bash
+   brew install coreutils jq
+   ```
+3. Ensure the newly installed tools are added to your `PATH`:
+   ```bash
+   echo 'export PATH="$PATH:/opt/homebrew/bin/:/usr/local/opt/coreutils/libexec/gnubin"' >> "${HOME}/.bashrc"
```

Comment:
I think it should still work since the commands that require these tools are called via `bash ...`.

I will test the instructions before merging this PR, though.